appender.go

```diff
@@ -9,24 +9,46 @@ import (
 	"database/sql/driver"
 	"errors"
 	"fmt"
+	"reflect"
 	"time"
 	"unsafe"
 )
 
-// Appender holds the DuckDB appender. It allows to load bulk data into a DuckDB database.
+type LogicalTypeEnum int
+
+const (
+	PRIMITIVE LogicalTypeEnum = iota
+	LIST
+	STRUCT
+)
+
+type SetColumnValue func(a *Appender, columnInfo *ColumnInfo, rowIdx C.idx_t, val interface{})
+
+type ColumnInfo struct {
+	vector   C.duckdb_vector
+	function SetColumnValue
+
+	colType LogicalTypeEnum
+
+	// Only for nested types so that we can recursively store the child column info
+	fields      int // the number of fields in the struct
+	columnInfos []ColumnInfo
+}
+
+// Appender holds the DuckDB appender. It allows data bulk loading into a DuckDB database.
```

Comment:
Nit: I am tempted to change this comment to something like this:
> Appender holds the DuckDB appender. It allows efficient bulk loading into a DuckDB database.

People should use the `Appender` for bulk loading. 😄